packages/@aws-cdk/aws-iotevents/README.md

```diff
@@ -40,65 +40,209 @@ Import it into your code:
+
+declare const input: iotevents.IInput;
+
+const setTemperatureAction = {
+  bind: () => ({
+    configuration: {
+      setVariable: { 
+        variableName: 'temperature', 
+        value: iotevents.Expression.inputAttribute(input, 'payload.temperature').evaluate(),
+      },
+    },
+  }),
+};
+
+const state = new iotevents.State({
+  stateName: 'MyState',
+  onEnter: [{ // optional
+    eventName: 'onEnter',
+    actions: [setTemperatureAction], // optional
+    condition: iotevents.Expression.currentInput(input), // optional
+  }],
 });
+```
+
+Also you can use `onInput` and `onExit`. `onInput` is triggered when messages are put to the input
+that is refered from the detector model. `onExit` is triggered when exiting this state.
 
-// transit to coldState when temperature is 10
-warmState.transitionTo(coldState, {
+```ts
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+const state = new iotevents.State({
+  stateName: 'warm',
+  onEnter: [{ // optional
+    eventName: 'onEnter',
+  }],
+  onInput: [{ // optional
+    eventName: 'onInput',
+  }],
+  onExit: [{ // optional
+    eventName: 'onExit',
+  }],
+});
+```
+
+You can set transitions of the states as following:
+
+```ts
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+declare const input: iotevents.IInput;
+declare const action: iotevents.IAction;
+declare const stateA: iotevents.State;
+declare const stateB: iotevents.State;
+
+// transit from stateA to stateB when temperature is 10
+stateA.transitionTo(stateB, {
   eventName: 'to_coldState', // optional property, default by combining the names of the States
+  actions: [action], // optional,
   when: iotevents.Expression.eq(
     iotevents.Expression.inputAttribute(input, 'payload.temperature'),
     iotevents.Expression.fromString('10'),
   ),
 });
-// transit to warmState when temperature is 20
-coldState.transitionTo(warmState, {
-  when: iotevents.Expression.eq(
-    iotevents.Expression.inputAttribute(input, 'payload.temperature'),
-    iotevents.Expression.fromString('20'),
-  ),
-});
+```
+
+## `DetectorModel`
+
+You can create `DetectorModel` as following.
+
+```ts
+import * as iotevents from '@aws-cdk/aws-iotevents';
+
+declare const state: iotevents.State;
 
 new iotevents.DetectorModel(this, 'MyDetectorModel', {
   detectorModelName: 'test-detector-model', // optional
   description: 'test-detector-model-description', // optional property, default is none
   evaluationMethod: iotevents.EventEvaluation.SERIAL, // optional property, default is iotevents.EventEvaluation.BATCH
   detectorKey: 'payload.deviceId', // optional property, default is none and single detector instance will be created and all inputs will be routed to it
-  initialState: warmState,
+  initialState: state,
 });
 ```
 
-To grant permissions to put messages in the input,
-you can use the `grantWrite()` method:
+## Examples
+
+The following example creates an AWS IoT Events detector model to your stack.
```

Comment:
That's great! I was just thinking that I would like to post some of [these examples](https://docs.aws.amazon.com/iotevents/latest/developerguide/iotevents-examples.html) as well.
I will move this example to `aws-iotevents-actions` because finally this example use actions😉.